### ISSUE TITLE
Proposal: Add threshold to prefer-destructuring based on number of extracted values

### ISSUE THREAD

**Issue by luchsamapparat** [eslint/eslint#8113]
**What rule do you want to change?**
`prefer-destructuring`

**Does this change cause the rule to produce more or fewer warnings?**
fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new `number` typed option called `minExtractedProperties` is added to the second configuration object of the rule. Feel free to propose a better name for that option, this was just the first that came to my mind :-)

**Please provide some example code that this change will affect:**

```js
const obj = {
    one: 1,
    two: 2,
    three: 3
};

const one = obj.one;
```

**What does the rule currently do for this code?**
It throws a warning: `Use object destructuring (prefer-destructuring)`

**What will the rule do after it's changed?**
It does not throw a warning if the extracted properties are below the `minExtractedProperties` threshold.

I think the destructuring syntax is overkill and not very readable if you just want to extract one single property. However, I would like to enforce destructuring when more than one property is being extracted.

**Comment by platinumazure** [comment]
Vaguely related proposal based on destructuring depth: #7973

Just wanted to link together all the proposals trying to solve the problem of "too much destructuring".

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
